src/features/accounts/index.tsx

```diff
@@ -1,26 +1,49 @@
-import { Avatar, ListItemAvatar } from "@mui/material";
+import { Avatar, Button, ListItemAvatar } from "@mui/material";
 import { AccountDetailsDto, AccountImportDto } from "lib/network/swagger-client";
 import { useRequestContext } from "providers/request-provider";
 import { AccountListItem, AccountListItemText } from "./index.styled";
 import {
-  accountListBreadcrumbLinks,
+  accountGridSettingsStorageKey,
   accountListCurrentBreadcrumb,
   defaultFilterOrderColumn,
   defaultFilterOrderDirection,
   modelName,
   searchLabel,
 } from "./constants";
-import { DataList } from "@components/data-list-old";
+import { DataList } from "@components/data-list";
 import { GridColDef } from "@mui/x-data-grid";
-import { CoreModule } from "lib/router";
+import { CoreModule, getAddFormRoute } from "lib/router";
+import { ModuleWrapper } from "@components/module-wrapper";
+import { dataListBreadcrumbLinks } from "utils/constants";
+import { SearchBar } from "@components/search-bar";
+import { Fragment, useRef, useState } from "react";
+import { Add, Download, Upload } from "@mui/icons-material";
+import { CsvImport } from "@components/spreadsheet-import";
+import { CsvExport } from "@components/export";
+import useLocalStorage from "use-local-storage";
+import { dataListSettings } from "utils/types";
+import { getModelByName } from "@lib/network/swagger-models";
+import { Result } from "@wavepoint/react-spreadsheet-import/types/types";
+import { GhostLink } from "@components/ghost-link";
 
 export const Accounts = () => {
   const { client } = useRequestContext();
+  const [gridSettings, setGridSettings] = useLocalStorage<dataListSettings | undefined>(
+    accountGridSettingsStorageKey,
+    undefined
+  );
+
+  const [searchTerm, setSearchTerm] = useState(gridSettings?.searchTerm ?? "");
+  const [openImport, setOpenImport] = useState(false);
+  const [openExport, setOpenExport] = useState(false);
+  const [importFieldsObject, setImportFieldsObject] = useState<any>();
+  const dataExportQuery = useRef("");
```

Comment:
why use ref?

Reply:
Child component (grid) will call parent's getAccountList method to load data list and at the same time it will build the export query.
To keep the export query, if we use a state variable with useState, it will cause a re-render of parent components which is unnecessary at that moment. That's why useRef is used here.